### DISCUSSION TITLE
Why is DynamoDB not a suitable backend datastore?

### DISCUSSION THREAD

**Discussion by TheHolyWaffle** [rocicorp/replicache#1061] [General] [answered]
The docs mention that AWS DynamoDB is not a suitable backend datastore, because it does not have "snapshot isolation". 

I'm surprised by this because AWS DynamoDB defaults to eventual consistency, but you can override this to be strongly consistent. 

See: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadConsistency.html

> Strongly Consistent Reads
Read operations such as GetItem, Query, and Scan provide an optional ConsistentRead parameter. If you set ConsistentRead to true, DynamoDB returns a response with the most up-to-date data, reflecting the updates from all prior write operations that were successful. Strongly consistent reads are only supported on tables and local secondary indexes. Strongly consistent reads from a global secondary index or a DynamoDB stream are not supported.

Is using this `ConsistentRead` parameter not good enough for Replicache's backend datastore requirement?

**Comment by aboodman** [comment]
There are some good threads on this on the discord. See for example:

https://discord.com/channels/830183651022471199/1246831723958763573/1247148510382391326

Although Dynamo does support 'consistent reads' it doesn't have enough features when using consistent reads to implement the pull endpoint.

**Comment by aboodman** [comment, marked as answer]
Here's another one: https://discord.com/channels/830183651022471199/830183651022471202/1156488579736944692

**Comment by aboodman** [comment]
hope this helps!
